File: 2559_수열.py
import sys
input = sys.stdin.readline

n, k = map(int, input().split())
arr = list(map(int, input().split()))

dp = [-1e9 for _ in range(n)]
dp[0] = arr[0]
for i in range(1, k):
    dp[i] = arr[i] + dp[i-1]

for i in range(k, n):
    dp[i] = arr[i] + dp[i-1] - arr[i-k]

print(max(dp[k-1:]))

File: src/main/kotlin/boj/implementation/2559_수열.kt
package boj.implementation

/**
 * 문제 번호: (2559)
 * 문제 이름: (수열)
 * 소요 시간(분): (12)
 */

import kotlin.math.max

fun main() = with(System.`in`.bufferedReader()) {
    val (n, k) = readLine().split(" ").map { it.toInt() }
    val arr = readLine().split(" ").map { it.toInt() }
    val pArr = IntArray(n + 1)
    for (i in 1 until n + 1) {
        pArr[i] = arr[i - 1] + pArr[i - 1]
    }

    var result = Int.MIN_VALUE
    for (i in k until n + 1) {
        result = max(result, pArr[i] - pArr[i - k])
    }
    println(result)
    close()
}